pandas/core/groupby/groupby.py

```diff
@@ -689,7 +689,22 @@ def _set_result_index_ordered(self, result):
             result.set_axis(index, axis=self.axis, inplace=True)
             result = result.sort_index(axis=self.axis)
 
-        result.set_axis(self.obj._get_axis(self.axis), axis=self.axis, inplace=True)
+        return self._restore_grouper_index(result)
+
+    def _restore_grouper_index(self, result: FrameOrSeries) -> FrameOrSeries:
```

Comment:
i would not create a new function, just push  this into _set_result_index_ordered. as its part of this